src/adapter/src/telemetry.rs

```diff
@@ -62,11 +64,15 @@ impl SegmentClientExt for mz_segment::Client {
                 json!(environment_id.cloud_provider_region()),
             );
         }
-        self.track(
-            user_id,
-            event,
-            properties,
-            Some(json!({ "group_id": environment_id.organization_id() })),
-        );
+        let app = json!({ "name": app_name });
+
+        // "Context" is a defined dictionary of extra information related to a datapoint. Please
+        // consult the docs before adding anything here: https://segment.com/docs/connections/spec/common/#context
+        let context = json!({
+            "app": app,
```

Comment:
I totally see where you're coming from, the reason I opted to use `app` is because I looked at `environmentd` as sort of an "implementation detail", and the actual application making the user request to Materialize being described with `application_name`. This is a bit of a stretch though and what's better probably better aligned with this description `userAgent`, although I don't think we should re-use that for this purpose since `userAgent` has specific ties to web browsers.

Given how thorough the Segment spec is, I wanted to try and reuse some of it, but I think you're right, the best option here is to add a new `application_name` to the event properties itself!